### PR TITLE
build: update infrastructure

### PR DESCRIPTION
**Pull request by abhijeetnarvekar**

### CODE REVIEW

File: src/index.css [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-html, body {
+ html, body {
```
Comment (thilo-m) [inline src/index.css]: why the added trailing space here?

File: tests/select-ranges.test.ts [hunk 2]
```diff
@@ -81,39 +82,42 @@ describe('live demo', () => {
...
-      await driver.wait(driver.findElement(MARKING_LOCATOR));
+      await driver.wait(driver.findElement(By.css("." + MARKING_CSS_CLASS)));
       await screenShooter.shoot('after-extract-text');
     });
 
-    it('display the extracted text', async () => {
-      const markingsParent = await driver.findElement(MARKING_LOCATOR).findElement(By.xpath('..'));
-      expect(await markingsParent.getText()).toEqual(TEST_TEXT);
+    test('display the extracted text', async () => {
+      const markingsParent = await driver.findElement(By.css("." + MARKING_CSS_CLASS)).findElement(By.xpath('..'));
```
Comment (thilo-m) [inline tests/select-ranges.test.ts]: I think we can omit the `"." +` here and in line 90 below too

Comment (abhijeetnarvekar) [reply tests/select-ranges.test.ts]: Fixed: Used marking locator which is readily defined

File: src/index.css (hunk 1 above)
Comment (abhijeetnarvekar) [reply src/index.css]: Fixed: Copied prettier config from sdk-js and applied it to entire project

File: package.json [hunk 3]
```diff
@@ -1,73 +1,48 @@
...
+    "eslint": "^8.57.0",
+    "eslint-plugin-react-hooks": "^4.6.2",
+    "eslint-plugin-react-refresh": "^0.4.7",
+    "fs-extra": "^11.2.0",
+    "jasmine": "^5.1.0",
+    "prettier": "^3.3.2",
+    "selenium-webdriver": "^4.22.0",
+    "typescript": "^5.5.3",
+    "vite": "^5.3.3",
+    "vitest": "^2.0.1"
```
Comment (thilo-m) [inline package.json]: latest version is 2.0.2

File: package.json [hunk 4]
```diff
@@ -1,73 +1,48 @@
...
+    "@types/selenium-webdriver": "^4.1.24",
+    "@typescript-eslint/eslint-plugin": "^7.13.1",
+    "@typescript-eslint/parser": "^7.13.1",
+    "@vitejs/plugin-react": "^4.3.1",
+    "browser-sync": "^3.0.2",
+    "chromedriver": "^126.0.4",
+    "dotenv": "^16.4.5",
+    "eslint": "^8.57.0",
+    "eslint-plugin-react-hooks": "^4.6.2",
+    "eslint-plugin-react-refresh": "^0.4.7",
```
Comment (thilo-m) [inline package.json]: latest version is 0.4.8

File: package.json [hunk 5]
```diff
@@ -1,73 +1,48 @@
...
+    "@types/react": "^18.3.3",
+    "@types/react-dom": "^18.3.0",
+    "@types/selenium-webdriver": "^4.1.24",
+    "@typescript-eslint/eslint-plugin": "^7.13.1",
+    "@typescript-eslint/parser": "^7.13.1",
+    "@vitejs/plugin-react": "^4.3.1",
+    "browser-sync": "^3.0.2",
+    "chromedriver": "^126.0.4",
+    "dotenv": "^16.4.5",
+    "eslint": "^8.57.0",
```
Comment (thilo-m) [inline package.json:37]: latest version is 9.6.0, could we use that?

File: .github/dependabot.yml
Comment (thilo-m) [inline .github/dependabot.yml:1]: Single and double quotes probably both work fine, but I should mention that we have basically copied this file into every GitHub/GitLab repository, using double quotes. So if we start to deviate here in one repo, that is a bit odd. Maybe revert in favor of consistency?

Comment (abhijeetnarvekar) [reply .github/dependabot.yml:1]: Reverted

File: package.json (hunk 5 above)
Comment (abhijeetnarvekar) [reply package.json:37]: Lets do the eslint as another PR

File: package.json (hunk 4 above)
Comment (abhijeetnarvekar) [reply package.json]: fixed: updated

File: package.json (hunk 3 above)
Comment (abhijeetnarvekar) [reply package.json]: Fixed: updated

File: .github/workflows/dependabot-auto-merge.yml [hunk 6]
```diff
@@ -14,7 +14,7 @@ jobs:
         id: metadata
         uses: dependabot/fetch-metadata@v2
         with:
-          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          github-token: '${{ secrets.GITHUB_TOKEN }}'
```
Comment (thilo-m) [inline .github/workflows/dependabot-auto-merge.yml]: Revert this file too?

Comment (abhijeetnarvekar) [reply .github/workflows/dependabot-auto-merge.yml]: reverted

**Review by thilo-m — APPROVED** [review]

**Pull request merged**